src/rust/include/rust/orchard.h

```diff
@@ -161,8 +161,8 @@ class AuthValidator
 
     /// Creates a validation context that performs no validation. This can be
     /// used when avoiding duplicate effort such as during reindexing.
-    static AuthValidator Disabled() {
-        return AuthValidator();
+    static std::optional<AuthValidator> Disabled() {
```

Comment:
I thought about removing this entirely, but c++ can't infer that `std::nullopt` conforms to a function argument of type `std::optional<AuthValidator>&`; this requires fewer changes to the tests but we might want to revisit it later.